Presentation/PresentationStyle.swift

```diff
@@ -274,3 +289,82 @@ private class PopoverPresentationControllerDelegate: NSObject, UIPopoverPresenta
 }
 
 private var isIpad: Bool { return UIDevice.current.userInterfaceIdiom == .pad }
+
+private class AdaptiveProxyPresentationDelegate: NSObject, UIAdaptivePresentationControllerDelegate {
+    private let didDismissCallbacker = Callbacker<()>()
```

Comment:
After some more reading of the code and hopefully a better understanding I wonder if it is not cleaner to make AdaptiveProxyPresentationDelegate public with support for all the delegates signals and delegates similar to tableViewDelegate and friends. Then we can check when modally presenting if the delegate is nil or of type AdaptiveProxyPresentationDelegate and use the delegate to handle dismiss, or else not handle dismiss (as someone else is doing that instead) and potentially log a warning.

Reply:
@mansbernhardt we did some changes based on this feedback. Unfortunately, we can't allow the user's to set directly the delegate to the `presentationController` of the vc because it causes memory leak problems when trying to access the `presentationController` before it gets presented on the screen. Instead we keep the custom delegate on the VC and the user needs to set that one instead so that Presentation can set it properly when we are presenting the vc

Reply:
Not sure I follow? Who is leaking? The one who is setting the delegate needs to hold on to the delegate as well using e.g. `bag.hold(delegate)`